### ISSUE TITLE
Cannot read property 'map' of undefined

### ISSUE THREAD

**Issue by robertmain** [hacksalot/HackMyResume#31]
Whenever I run `hackmyresume build resume.json` I get an error saying `Cannot read property 'map' of undefined`. I feel like I'm missing something really obvious...

**Comment by hacksalot** [comment]
Thanks for the info -- have had one report of this already and, assuming it's the same issue, a fix should be out shortly. As a sanity check, you can try running the same command against [Jane Q.'s resume](https://github.com/fluentdesk/jane-q-fullstacker/blob/master/resume/jane-resume.json) and I'll update here either way.

**Comment by thatRailsGuy** [comment]
I'm getting this as well, might be related:

ERROR: Cannot read property 'history' of undefined

**Comment by thatRailsGuy** [comment]
When I try with Jane Q.'s resume I get the following:

ERROR: Object #<Object> has no method 'parse'

**Comment by hacksalot** [comment]
If you're still seeing this error, trying upgrading (via npm install/update) to 1.2.1 (latest as of today) or 1.3.0 (latest as of later today/tomorrow/after Christmas). Both versions include related fixes.

@thatRailsGuy Jane Q [updated her resume](https://github.com/fluentdesk/jane-q-fullstacker/commit/48ffdc1113c53b02a2b179f7e560d672db06e8f6) last night, if this error persists, grab latest as a sanity check. Thanks.

**Comment by Furchin** [comment]
+1 also seeing this issue, both with `HackMyResume` and `fluentcv`. JSON Resume format is deemed valid by both validators.

**Comment by Furchin** [comment]
FWIW the stack is 

```
TypeError: Cannot call method 'map' of undefined
    at writing (/usr/local/lib/node_modules/hackmyresume/src/core/convert.js:300:18)
    at Object.module.exports.toFRESH (/usr/local/lib/node_modules/hackmyresume/src/core/convert.js:56:18)
    at FreshResume.parse (/usr/local/lib/node_modules/hackmyresume/src/core/fresh-resume.js:157:23)
    at FreshResume.open (/usr/local/lib/node_modules/hackmyresume/src/core/fresh-resume.js:36:17)
    at /usr/local/lib/node_modules/hackmyresume/src/core/load-source-resumes.js:9:53
    at Array.map (native)
    at loadSourceResumes (/usr/local/lib/node_modules/hackmyresume/src/core/load-source-resumes.js:6:16)
    at generate (/usr/local/lib/node_modules/hackmyresume/src/verbs/generate.js:40:18)
    at main (/usr/local/lib/node_modules/hackmyresume/src/index.js:75:22)
    at Object.<anonymous> (/usr/local/lib/node_modules/hackmyresume/src/index.js:22:3)
```

I realize it might hurt the clean UI a bit, but I might recommend adding the stack trace to the `handleError` output, like:

```
f( !ex.fluenterror || ex.fluenterror < 3 ) {
127     console.log( ('ERROR: ' + trimmed.toString()).red.bold);
128     console.log( ex.stack.gray);
129   } else
130     console.log( trimmed.toString() );
```

**Comment by Furchin** [comment]
Getting the latest [Jane Q Fullstacker resume](https://github.com/fluentdesk/jane-q-fullstacker/blob/master/resume/jane-resume.json) and running hackmyresume@1.2.1 on it I'm seeing the same parse error as @thatRailsGuy:

```
TypeError: Object #<Object> has no method 'parse'
    at Theme.open (/usr/local/lib/node_modules/hackmyresume/src/core/theme.js:33:25)
    at generate (/usr/local/lib/node_modules/hackmyresume/src/verbs/generate.js:63:38)
    at main (/usr/local/lib/node_modules/hackmyresume/src/index.js:75:22)
    at Object.<anonymous> (/usr/local/lib/node_modules/hackmyresume/src/index.js:22:3)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
```

**Comment by yunesj** [comment]
"map" error with 1.2.1 on validated resume.

**Comment by hacksalot** [comment]
Working a fix for this now...

@Furchin Agreed, the error output is a little stingy! Will add in next bump.

**Comment by hacksalot** [comment]
Some occurrences of this and #31 possibly related to Node.js version issue -- [older versions don't support path.parse](http://stackoverflow.com/a/28878384).

**Comment by Furchin** [comment]
I'm reproing using Node `v0.10.29`. The Jane Q resume works on latest stable node (`5.2.0`).

**Comment by hacksalot** [comment]
Yay. Thanks for that and the call stack @Furchin. Fix in progress.

**Comment by Quantumplation** [comment]
I was running this in a docker container based off https://hub.docker.com/r/madkom/node-npm/.  But after switching to the official node container, this works great, thanks!

**Comment by hacksalot** [comment]
Please try v1.2.2 for those who are experiencing these issue. Thanks!

**Comment by petersandor** [comment]
That fixed it for me, thanks!

**Comment by Furchin** [comment]
Confirmed fixed here as well, thanks!

**Comment by robertmain** [comment]
Can't currently confirm this as fixed because I'm running into `Cannot read property start of undefined` instead now...

**Comment by hacksalot** [comment]
@Furchin, @Quantumplation, @thatRailsGuy, @robertmain Yay. Thanks for the reports.

@robertmain Thanks and this will be fixed this evening. In the meantime, as a potential workaround, it looks like you're missing a `start` date or one of the required resume sections. In v1.3 you'll be able to omit this stuff, but in v1.2.x it's still a little finnicky. Are you able to build any of the sample resumes?

**Comment by robertmain** [comment]
@hacksalot wow, thanks for grabbing this so soon(Merry Christmas! :smile: ). I'll try building one of the sample resumes(where do I find them?). I'm guessing the reason it's unhappy about the missing dates is because I removed the start and end dates from some of my volunteer work(it was at a one day conference, so the start and end dates seemed somewhat redundant)

**Comment by hacksalot** [comment]
@robertmain You're welcome and happy holidays! :snowman: 

The second part of the fix is in progress and as of v1.3 you'll be able to omit dates and other fields across the board, and if a field is required, a validation error will trigger. As for the samples, there are a few, but the FRESH sample is [here](https://github.com/fluentdesk/jane-q-fullstacker/blob/master/resume/jane-resume.json) and the JSON Resume sample is [here](https://github.com/jsonresume/resume-schema/blob/master/resume.json). v1.3 should be out today or tomorrow at latest.

**Comment by hacksalot** [comment]
If anybody is still seeing this issue in [v1.3.0](https://github.com/hacksalot/HackMyResume/releases/tag/v1.3.0) please log a new issue as the original issue(s) have either been resolved or revealed new issues. Thanks all.
